pkg/core/writer/correlations/client.go

```diff
@@ -0,0 +1,227 @@
+package correlations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/core/writer/requests"
+	log "github.com/sirupsen/logrus"
+)
+
+const correlationChanCapacity = 1000
+
+// CorrelationClient is an interface for correlations.Client
+type CorrelationClient interface {
+	Correlate(correlation *Correlation)
+	Delete(correlation *Correlation)
+	Get(dimName string, dimValue string, callback func(map[string][]string, error))
+	Start()
+}
+
+// NOOPClient implements CorrelationClient interface but doesn't do anything with the correlation
+type NOOPClient struct{}
```

Comment:
Put this in a separate file to declutter it from the core logic.

Reply:
Removed because we don't need now that we're removing the traceHostCorrelation config option